tempo_embeddings/text/abstractcorpus.py

```diff
@@ -0,0 +1,296 @@
+from abc import ABC
+from abc import abstractmethod
+from collections import Counter
+from collections import defaultdict
+from functools import lru_cache
+from typing import Any
+from typing import Iterable
+from typing import Optional
+import numpy as np
+from numpy.typing import ArrayLike
+from scipy.sparse import csr_matrix
+from sklearn.cluster import HDBSCAN
+from sklearn.feature_extraction.text import TfidfVectorizer
+from .highlighting import Highlighting
+from .passage import Passage
+
+
+class AbstractCorpus(ABC):
+    @property
+    @abstractmethod
+    def passages(self) -> list[Passage]:
+        return NotImplemented
+
+    @property
+    @abstractmethod
+    def embeddings(self) -> Optional[ArrayLike]:
+        return NotImplemented
+
+    @property
+    def label(self) -> Optional[str]:
+        return self._label
+
+    @label.setter
+    def label(self, value: str):
+        self._label = value
+
+    def __eq__(self, other: object) -> bool:
+        return other._passages == self._passages
```

Comment:
Yes, if the order is different, something is different. It should not be relevant in most cases, because `Corpus` equality is mostly used in tests anyway.